Src/FluentAssertions/Equivalency/Node.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using FluentAssertions.Common;
+
+namespace FluentAssertions.Equivalency
+{
+    public class Node : INode
+    {
+        public GetSubjectId GetSubjectId { get; protected set; } = () => string.Empty;
+
+        public Type Type { get; protected set; }
+
+        public string Path { get; protected set; }
+
+        public string PathAndName => Path.Combine(Name);
+
+        public string Name { get; protected set; }
+
+        public virtual string Description => $"{GetSubjectId().Combine(PathAndName)} (of type {Type.ToFriendlyName()})";
+
+        public virtual bool IsRoot => Path.Length == 0;
+
+        public static INode From<T>(GetSubjectId getSubjectId)
+        {
+            return new Node
+            {
+                GetSubjectId = () => getSubjectId() ?? "root",
+                Name = string.Empty,
+                Path = string.Empty,
+                Type = typeof(T),
+            };
+        }
+
+        public static INode FromCollectionItem<T>(string index, INode parent)
+        {
+            return new Node
+            {
+                Type = typeof(T),
+                Name = "[" + index + "]",
+                Path = parent.PathAndName,
+                GetSubjectId = parent.GetSubjectId
+            };
+        }
+
+        public static INode FromDictionaryItem<T>(object key, INode parent)
+        {
+            return new Node
+            {
+                Type = typeof(T),
+                Name = "[" + key + "]",
+                Path = parent.PathAndName,
+                GetSubjectId = parent.GetSubjectId
+            };
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, obj))
+            {
+                return true;
+            }
+
+            if (obj.GetType() != this.GetType())
+            {
+                return false;
+            }
+
+            return Equals((Node)obj);
+        }
+
+        private bool Equals(Node other) => Type == other.Type && PathAndName == other.PathAndName;
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+#if NETCOREAPP3_0
```

Comment:
I think we should just silence the analyzer here.
.netstandard2.1 and .netcoreapp2.1 also have the `GetHashCode(StringComparison)` overload, but the analyzer is only enabled for the .netcoreapp3.0 target.

Reply:
Unfortunately .NET Standard 2.0 does not.